### DISCUSSION TITLE
AppEngine: ModuleNotFoundError: No module named 'uvloop'

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#6790] [Questions] [answered]
Similar to issue #2181 the logs in the GCP states, that there is no module named uvloop. As someone have suggested to add a line of code in the pyproject.toml file, we use app.yaml files for app engine. What are we supposed to do there as we can't run uvicorn[standard] the same way as we did during the development phase?

**Comment by ghost** [comment]
I am receiving an error 502 due to this.

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/37276987/115512919-c89b4800-a282-11eb-83e6-f480df4f54f8.png)

**Comment by frankie567** [comment, marked as answer]
App Engine automatically detects and installs dependencies set in a classic **requirements.txt** file (see https://cloud.google.com/appengine/docs/standard/python3/specifying-dependencies).

You have to create such file where you list all your dependencies, like this (pin according to your own versions):

```txt
fastapi==0.63.0
uvicorn[standard]==0.13.4
```

**Comment by ghost** [comment]
I forgot to add the [standard] to the uvicorn and I forgot to get back here with my results. I wasn't exactly sure that the system would detect the square syntax but it worked. Thanks for the assist!

**Comment by frankie567** [comment]
You're welcome! As a side note, the square brackets syntax is standard in Python to specify optional dependencies, so it usually work everywhere: https://www.python.org/dev/peps/pep-0508/#extras

**Comment by ghost** [comment]
I see. That's great to hear. I am still very new to this (got a new job recently where I had to learn Python, so I've only got three/four months of experience with it so far. I used to be a Java guy 😄

**Comment by tiangolo** [comment]
Thanks for the help here @frankie567 ! 👏  🙇 

Thanks for reporting back and closing the issue @dia 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
